packages/react/menu/src/Menu.tsx

```diff
@@ -746,9 +746,16 @@ const MenuItemImpl = React.forwardRef<MenuItemImplElement, MenuItemImplProps>(
 const CHECKBOX_ITEM_NAME = 'MenuCheckboxItem';
 
 type MenuCheckboxItemElement = MenuItemElement;
+
+type CheckedState = boolean | 'indeterminate';
+
+function isIndeterminate(checked?: CheckedState): checked is 'indeterminate' {
+  return checked === 'indeterminate';
+}
```

Comment:
nit: Could we move this at the bottom, just between `getOpenState` and `getCheckedState`?